### ISSUE TITLE
`InlineUIContainer` dosen't correctly render with `TextProperty=Wrap`

### ISSUE THREAD

**Issue by Danich27rus** [AvaloniaUI/Avalonia#17183]
### Describe the bug

Right now if Inlines have InlineUIContainer inside and we have TextProperty=Wrap on TextBlock, everything is fine until InlineUIContainer renders.
![image](https://github.com/user-attachments/assets/1f812058-22b8-4501-ab2b-d8c87ed3127e)
![image](https://github.com/user-attachments/assets/a9b7ecc5-dcf7-4527-bdbb-5487da62691d)

After that, if i back to smaller width, InlineUiContainer will be rendered incorrectly, on top of my text.
![image](https://github.com/user-attachments/assets/766292b4-6bab-4fae-be85-78a936be0700)

I don't know what to do right now. I thought about using WrapPanel, but that means i need to rewrote whole system in my App.

### To Reproduce

1. Create inline.
2. Add runs and InlineUIContainer.
3. Set inlines on some TextBlock with changable width.
4. Try to change size on TextBlock
5. InlineUIContainer will show and never hide.

### Expected behavior

1. Create inline.
2. Add runs and InlineUIContainer.
3. Set inlines on some TextBlock with changable width.
4. Try to change size on TextBlock
5. InlineUIContainer will proper show and hide as width changes.

### Avalonia version

11.1.3

### OS

Windows

**Comment by Danich27rus** [comment]
Looks like this code
```
                            if (drawable is EmbeddedControlRun controlRun
                                && controlRun.Control is Control control)
                            {
                                control.Arrange(
                                    new Rect(new Point(currentX, currentY),
                                    new Size(control.DesiredSize.Width, textLine.Height)));
                                _aabb.Add(control);
                            }
```
in ArrangeOverride dosen't rearrange control when it calls again. I think it can be solved if there will be some kind of pool of all controls. I can do a PR, but i don't think store list of controls is a good thing.

**Comment by Danich27rus** [comment]
Created super simple fix that was written in 2 minutes:
```
List<Control> _aabb = new();

protected override Size ArrangeOverride(Size finalSize)
{
    var scale = LayoutHelper.GetLayoutScale(this);
    var padding = LayoutHelper.RoundLayoutThickness(Padding, scale, scale);

    //Fixes: #11019
    if (finalSize.Width < _constraint.Width)
    {
        _textLayout?.Dispose();
        _textLayout = null;
        _constraint = finalSize.Deflate(padding);
    }

    if (HasComplexContent)
    {
        foreach (var cont in _aabb)
        {
            cont.Arrange(
               new Rect(new Point(0, 0),
               new Size(0, 0)));
        }

        var currentY = padding.Top;

        foreach (var textLine in TextLayout.TextLines)
        {
            var currentX = padding.Left + textLine.Start;

            foreach (var run in textLine.TextRuns)
            {
                if (run is DrawableTextRun drawable)
                {
                    if (drawable is EmbeddedControlRun controlRun
                        && controlRun.Control is Control control)
                    {
                        control.Arrange(
                            new Rect(new Point(currentX, currentY),
                            new Size(control.DesiredSize.Width, textLine.Height)));
                        _aabb.Add(control);
                    }

                    currentX += drawable.Size.Width;
                }
            }

            currentY += textLine.Height;
        }
    }

    return finalSize;
}
```

Works as i wanted.
